crates/ruff_linter/src/rules/pylint/rules/unnecessary_list_index_lookup.rs

```diff
@@ -61,220 +181,84 @@ impl<'a> SubscriptVisitor<'a> {
-        let mut diagnostic = Diagnostic::new(UnnecessaryListIndexLookup, range);
-
-        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
-            value_name.clone(),
-            range,
-        )));
-
-        checker.diagnostics.push(diagnostic);
-    }
-}
 
-/// PLR1736
-pub(crate) fn unnecessary_list_index_lookup_comprehension(checker: &mut Checker, expr: &Expr) {
-    match expr {
-        Expr::GeneratorExp(ast::ExprGeneratorExp {
-            elt, generators, ..
-        })
-        | Expr::DictComp(ast::ExprDictComp {
-            value: elt,
-            generators,
-            ..
-        })
-        | Expr::SetComp(ast::ExprSetComp {
-            elt, generators, ..
-        })
-        | Expr::ListComp(ast::ExprListComp {
-            elt, generators, ..
-        }) => {
-            for comp in generators {
-                let Some((sequence, index_name, value_name)) =
-                    enumerate_items(checker, &comp.iter, &comp.target)
-                else {
+    fn visit_expr(&mut self, expr: &Expr) {
+        if self.modified {
+            return;
+        }
+        match expr {
+            Expr::Subscript(ast::ExprSubscript {
+                value,
+                slice,
+                range,
+                ..
+            }) => {
+                let Expr::Name(ast::ExprName { id, .. }) = value.as_ref() else {
                     return;
                 };
-
-                let mut visitor = SubscriptVisitor::new(&sequence, &index_name);
-
-                visitor.visit_expr(elt.as_ref());
-
-                for range in visitor.diagnostic_ranges {
-                    let mut diagnostic = Diagnostic::new(UnnecessaryListIndexLookup, range);
-
-                    diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
-                        value_name.clone(),
-                        range,
-                    )));
-
-                    checker.diagnostics.push(diagnostic);
+                if id == self.sequence_name {
+                    let Expr::Name(ast::ExprName { id, .. }) = slice.as_ref() else {
+                        return;
+                    };
+                    if id == self.index_name {
+                        self.diagnostic_ranges.push(*range);
+                    }
                 }
             }
+            _ => visitor::walk_expr(self, expr),
         }
-        _ => (),
-    }
-}
-
-fn enumerate_items(
-    checker: &mut Checker,
-    call_expr: &Expr,
-    tuple_expr: &Expr,
-) -> Option<(String, String, String)> {
-    let ast::ExprCall {
-        func, arguments, ..
-    } = call_expr.as_call_expr()?;
-
-    // Check that the function is the `enumerate` builtin.
-    let Some(call_path) = checker.semantic().resolve_call_path(func.as_ref()) else {
-        return None;
-    };
-
-    match call_path.as_slice() {
-        ["", "enumerate"] => (),
-        ["builtins", "enumerate"] => (),
-        _ => return None,
     }
```

Comment:
I rewrote this as:

```rust
    // Check that the function is the `enumerate` builtin.
    if !semantic
        .resolve_call_path(func.as_ref())
        .is_some_and(|call_path| matches!(call_path.as_slice(), ["builtins" | "", "enumerate"]))
    {
        return None;
    }
```

Note, however, that you could also do `let call_path = checker.semantic().resolve_call_path(func.as_ref())?` here, which is more idiomatic than `let`-`else` with a `return None`.